src/utils/transitions-for.js

```diff
@@ -0,0 +1,40 @@
+import { reduceObject } from 'ioo';
+import mergeDeepRight from 'ramda/src/mergeDeepRight';
+
+import MicrostateObject from '../primitives/object';
+import getReducerType from './get-reducer-type';
+import getTypeDescriptors from './get-type-descriptors';
+import transition from './transition';
+
+export default function transitionsFor(Type) {
+  let type = getReducerType(Type);
+  let transitions = reduceObject(
+    getTypeDescriptors(type.prototype),
+    (accumulator, descriptor, name) => {
+      return Object.assign({}, accumulator, {
+        [name]: name === 'initialize' ? descriptor.value : transition(descriptor.value),
+      });
+    },
+    {
+      set: transition(function set(current, state) {
+        return state && state.valueOf ? state.valueOf() : state;
+      }),
+    }
+  );
+  let isComposed = type === Type;
```

Comment:
Is this necessary? If we've played our cards right, then the entire recursive structure should be captured in the `Functor` instance for tree. If we have to worry about that as we gather information about a node, then we've lost.

Reply:
I'll look into this